README.md

```diff
@@ -28,31 +28,134 @@ flag is highly recommended for those new to [pip](https://pip.pypa.io).
 
-## Development
+## Quick Start
 
-To contribute or develop with this library, see
-[CONTRIBUTING](https://github.com/planetlabs/planet-client-python/CONTRIBUTING.md)
+### Python Library
 
+The python library is asynchronous, which greatly speeds up many interactions
+with the planet API. 
 
-## API Key
+```python
+import asyncio
+import os
+from planet.api import APlanetSession, AOrdersClient
+from planet.api.order_details import OrderDetails, Product
 
-The API requires an account for use. [Signup here](https://www.planet.com/explorer/?signup).
+API_KEY = os.getenv('PL_API_KEY')
 
-This can be provided via the environment variable `PL_API_KEY` or the flag `-k` or `--api-key`.
+image_ids = ['3949357_1454705_2020-12-01_241c']
+order_detail = OrderDetails(
+    'test_order',
+    [Product(image_ids, 'analytic', 'psorthotile')]
+)
 
-Using `planet init` your account credentials (login/password) can be used to obtain the api key.
+async def create_order(order_detail):
+    async with APlanetSession(auth=(API_KEY, '')) as ps:
+        client = AOrdersClient(ps)
+        return await client.create_order(order_detail)
 
+oid = asyncio.run(create_order())
+print(oid)
+```
+
+Not into async? No problem. Just wrap the library and async operations together
+and call from your synchronous code.
 
-# Example CLI Usage
+```python
+def sync_create_order(order_details):
+    return asyncio.run(create_order())
+
+print(sync_create_order(order_details))
+```
+
+Why async? Because things get *really cool* when you want to work with multiple
+orders. Here's an example of submitting two orders, waiting for them to
+complete, and downloading them. The orders each clip a set of images to a
+specific area of interest (AOI), so they cannot be combined into one order.
+(hint: [Planet Explorer](https://www.planet.com/explorer/) was used to define
+the AOIs and get the image ids.)
+ 
+
+```python
+import asyncio
+import os
+from planet.api import APlanetSession, AOrdersClient
+from planet.api.order_details import OrderDetails, Product, Tool
+
+API_KEY = os.getenv('PL_API_KEY')
+
+iowa_aoi = {
+    "type": "Polygon",
+    "coordinates": [[
+        [-91.198465, 42.893071],
+        [-91.121931, 42.893071],
+        [-91.121931, 42.946205],
+        [-91.198465, 42.946205],
+        [-91.198465, 42.893071]]]
+}
+
+iowa_images = [
+    '20200925_161029_69_2223',
+    '20200925_161027_48_2223'
+]
+iowa_order = OrderDetails(
+    'iowa_order',
+    [Product(iowa_images, 'analytic', 'PSScene4Band')],
+    tools=[Tool('clip', {'aoi': iowa_aoi})]
+)
+
+oregon_aoi = {
+    "type": "Polygon",
+    "coordinates": [[
+        [-117.558734, 45.229745],
+        [-117.452447, 45.229745],
+        [-117.452447, 45.301865],
+        [-117.558734, 45.301865],
+        [-117.558734, 45.229745]]]
+}
+
+oregon_images = [
+    '20200909_182525_1014',
+    '20200909_182524_1014'
+]
+oregon_order = OrderDetails(
+    'oregon_order',
+    [Product(oregon_images, 'analytic', 'PSScene4Band')],
```

Comment:
I find building up these bespoke request objects to be a bit cumbersome. It would be great if the python client could provide some simplified abstractions (not necessarily in this already quite large PR).

As one example, we could have an order utility that accepts an AOI and TOI, does a search, and places an order.

Reply:
Agreed, and saved in #250. Given that we just have the orders client implemented now and don't have that higher level functionality, how is the feel of building up the request object this way? The old, JSON blob way is also supported. Is this an improvement over the JSON blob?

Reply:
Ok, in the interest of keeping this PR rolling I am going to mark this as resolved with the new issue. I am still interested in hearing if you see other opportunities for improvement. If you have thoughts, new issue with suggestions is *very* welcome!

Reply:
**TL;DR: I prefer dictionaries over classes for this sort of thing. :)**

I personally prefer the JSON blobs and dictionaries, to avoid the cost of JSON <--> `dict` <--> class as well as the slightly slower dot attribute lookup compared to the O(1) of a `dict` lookup. What I'm saying might be a premature optimization, but I like the parallel-ish UX of something like:

```shell
curl -X POST --u ${PL_API_KEY}: -H '<json>' <create-order-endpoint> -d @order.json
```

```shell
planet orders create --json=@order.json
```

```python
import planet

def read_json(path: str) -> str:
    with open(path, "r") as f:
        return f.read()

oid = planet.create_order(read_json("order.json"))
print(oid)
```

I feel like there might be two different use cases here, and I'm not sure how to address all the things, but there's one case where using classes for examples and stuff is super nice, but then there's the case where folks are spinning up millions of these things and pushing them at the API and walking away, and so the class-based ergonomics might not be as beneficial or valuable.

Here's an example of what I'm trying to communicate :)

```python
from __future__ import annotations
import dataclasses
import dis
import functools
import json
import timeit
import typing


@dataclasses.dataclass
class SerializableDataclass:

    @classmethod
    def fromdict(cls, data: dict) -> SerializableDataclass:
        return cls(**data)

    @classmethod
    def fromjson(cls, data: str) -> SerializableDataclass:
        return cls.fromdict(json.loads(data))

    def asjson(self) -> str:
        return json.dumps(dataclasses.asdict(self))


@dataclasses.dataclass
class Product(SerializableDataclass):
    image_ids: typing.List[str]
    item_type: str
    product_bundle: str


@dataclasses.dataclass
class Order(SerializableDataclass):
    name: str
    products: typing.List[Product]


order_json = """
    {
        "name": "test_order",
        "products": [{
            "image_ids": "3949357_1454705_2020",
            "item_type": "psorthotile",
            "product_bundle": "analytic"
        }]
    }
"""


def test_order_dict_name(order_json: str) -> str:
    order = json.loads(order_json)
    return order["name"]


def test_order_class_name(order_json: str) -> str:
    order = Order.fromjson(order_json)
    return order.name


def main():
    print(dis.dis(test_order_dict_name))
    print(dis.dis(test_order_class_name))
    timeme = functools.partial(timeit.timeit, number=10_000_000)
    dict_time = timeme(lambda: test_order_dict_name(order_json))
    cls_time = timeme(lambda: test_order_class_name(order_json))
    print(f"dict = {dict_time}")
    print(f"cls = {cls_time}")


if __name__ == "__main__":
    main()
```

And this is the output I got from one run of that:
```shell
 51           0 LOAD_GLOBAL              0 (json)
              2 LOAD_METHOD              1 (loads)
              4 LOAD_FAST                0 (order_json)
              6 CALL_METHOD              1
              8 STORE_FAST               1 (order)

 52          10 LOAD_FAST                1 (order)
             12 LOAD_CONST               1 ('name')
             14 BINARY_SUBSCR
             16 RETURN_VALUE
None
 56           0 LOAD_GLOBAL              0 (Order)
              2 LOAD_METHOD              1 (fromjson)
              4 LOAD_FAST                0 (order_json)
              6 CALL_METHOD              1
              8 STORE_FAST               1 (order)

 57          10 LOAD_FAST                1 (order)
             12 LOAD_ATTR                2 (name)
             14 RETURN_VALUE
None
dict = 31.057720659009647
cls = 44.18623552902136
```

And there's a bug in this because the `Product` class as a type of the `products` member of `Order` is still a list of `dict`s because only top-level deserialization is here. Thar be dragons! :)

Reply:
@ericrdunham this is some great feedback I am processing here. 

First, in mentioning the bug, are you talking about the code you've written here or in the client code?

Currently, the creation of an order can be done with a JSON blob as well as classes. That would address the user experience (note to self: document this better #242). But internally the JSON blob is converted into classes. So that will slow things down. If I am reading this right, the performance hit is 13 seconds in 10 million runs. That doesn't sound so bad but also that performance hit is between 1/3 and 1/2 of the optimized runtime. So that sounds a little worse. Seems worthwhile to at least put up a ticket to consider that optimization.

The ultimate goal with all of the `order_details` module is to help the user construct an order without having to parse all of the API docs (and cross-reference compatible product bundles and item types). I am very open to any ideas about different ways to achieve this!

Reply:
ok, I would love to continue discussion on how we can improve order specification in #254. It is valuable and also big enough to warrant it's own ticket and effort. For this PR I'll consider the work captured and thus resolved here.

Reply:
@jreiberkyle the bug is in the example code I wrote because it doesn't deserialize child `Product` class instances that are supposed to be in the `Order` instance. Instead of `typing.List[Product]`, they are effectively `typing.List[typing.Mapping[str, str]]` at the moment. :-/ 

The performance hit there was only for single key/attribute lookup, for what that's worth, and I don't know if that's representative of what a customer would want to do with it. :)

The question I suppose I'd have here is, would users want this? My first thought was to advocate for JSON and `dict`s over classes and say something about the Zen of Python like:
> There should be one-- and preferably only one --obvious way to do it.
... but I feel as though that's because of my workflows and performance concerns and how I've preferred to do things.

I'd like to chat about this asynchronously over e-mail or direct messages, and as context for that conversation I'd really like to see what user feedback we have about what folks would like to see in a library for this sort of thing and use more of that type of feedback to push for changes. Thoughts @jreiberkyle @strixcuriosus @SaraPlanet?